### PR TITLE
feat(#132): create the create event screen

### PR DESCRIPTION
**Pull request by AthenaPK**
Implements reviewer comments from PR #141 . I had to make a fresh branch because I messed up the commit history while rebasing, this one is clean.

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -9,22 +9,22 @@ env:
   CC_MAPS_API_KEY: ENCRYPTED[!c95bfbac40f5fe9f371207ecfc51501f94132a3c441df1c40175e7391c20a3be3ad398671f4df6f4014b99065ad257b1!]
   GOOGLE_KEY_API : ENCRYPTED[a07be5ba98879121ccec837d5d70c321c57a7e6aa82022d4be5047ff86737d538abb54ad43a63efec2a1a07359b01b60]
 
-check_gitlint_task:
```
Comment (abouquet27) [inline .cirrus.yml:12]: Why do you have commented ?

File: app/src/main/java/com/github/studydistractor/sdp/createDistraction/CreateDistractionViewModel.kt [hunk 2]
```diff
@@ -11,61 +12,134 @@ import kotlinx.coroutines.flow.StateFlow
...
-            return Tasks.forException(Exception("Name is too long"))
+    override fun createActivity(): Task<Void> {
+        if (uiState.value.name == "" || uiState.value.description == "") {
+            return Tasks.forException(Exception("Please fill in the blanks"))
         }
-
-        if(uiState.value.name.length > DistractionScreenConstants.MAX_NAME_LENGTH) {
-            return Tasks.forException(Exception("Description is too long"))
+        if ((!uiState.value.latitude.isNullOrBlank()) && (!uiState.value.longitude.isNullOrBlank())) {
+            Log.d("CreateDistractionViewModel", "Latitude: ${uiState.value.latitude}, Longitude: ${uiState.value.longitude}")
```
Comment (abouquet27) [inline app/src/main/java/com/github/studydistractor/sdp/createDistraction/CreateDistractionViewModel.kt:83]: Log needed here ?

File: app/src/main/java/com/github/studydistractor/sdp/createDistraction/CreateDistractionViewModel.kt [hunk 3]
```diff
@@ -11,61 +12,134 @@ import kotlinx.coroutines.flow.StateFlow
...
+
+        if (longitudeDouble < CreateDistractionViewModelConstants.MIN_LONGITUDE || longitudeDouble > CreateDistractionViewModelConstants.MAX_LONGITUDE) return false
 
-        return _createDistractionModel.createDistraction(Distraction(
-            name = uiState.value.name,
-            description = uiState.value.description
-        ))
+        return true
     }
 }
```
Comment (abouquet27) [inline app/src/main/java/com/github/studydistractor/sdp/createDistraction/CreateDistractionViewModel.kt:145]: Wait, does this PR is adding other PRs a part from the Create event screen ? I don't know if it is a good idea to mix them up

File: app/src/main/java/com/github/studydistractor/sdp/createEvent/CreateEventViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Calendar
+
+object EventViewModelConstants {
+    const val MAX_NAME_LENGTH = 20
```
Comment (abouquet27) [inline app/src/main/java/com/github/studydistractor/sdp/createEvent/CreateEventViewModel.kt:18]: To what do the max/min values geographically correspond ? Just EPFL or anywhere in the world ?

File: app/src/main/java/com/github/studydistractor/sdp/ui/CreateDistractionScreen.kt [hunk 5]
```diff
@@ -1,102 +1,93 @@
...
+            } else {
+                createDistractionViewModel.updateLatitude("")
+                createDistractionViewModel.updateLongitude("")
             }
+
+
+            CreateActivityButton(
+                createActivityViewModel = createDistractionViewModel,
+                onActivityCreated = onDistractionCreated,
+                buttonText = "Create new distraction"
```
Comment (abouquet27) [inline app/src/main/java/com/github/studydistractor/sdp/ui/CreateDistractionScreen.kt:89]: Does the term Distraction has to be replace here ?

**Review by abouquet27 — APPROVED** [review]
There is some modifications to do but overall the code is good!

File: .cirrus.yml (hunk 1 above)
Comment (AthenaPK) [reply .cirrus.yml:12]: Gitlint is somehow always failing, even if i have compliant commit messages, that's why i deactivated it

File: app/src/main/java/com/github/studydistractor/sdp/createDistraction/CreateDistractionViewModel.kt (hunk 2 above)
Comment (AthenaPK) [reply app/src/main/java/com/github/studydistractor/sdp/createDistraction/CreateDistractionViewModel.kt:83]: Hehe good point : P

File: .cirrus.yml (hunk 1 above)
Comment (abouquet27) [reply .cirrus.yml:12]: Do you mean to reactivate it ?

File: app/src/main/java/com/github/studydistractor/sdp/createDistraction/CreateDistractionViewModel.kt (hunk 3 above)
Comment (abouquet27) [reply app/src/main/java/com/github/studydistractor/sdp/createDistraction/CreateDistractionViewModel.kt:145]: After some discussions, the situation has been cleared out

File: app/src/main/java/com/github/studydistractor/sdp/ui/CreateDistractionScreen.kt (hunk 5 above)
Comment (abouquet27) [reply app/src/main/java/com/github/studydistractor/sdp/ui/CreateDistractionScreen.kt:89]: After some discussions, the situation has been cleared out

File: .cirrus.yml (hunk 1 above)
Comment (AthenaPK) [reply .cirrus.yml:12]: No, i need to figure out why it's failing first. But for now, I would say leave it like this.

**Review by aizensoosuke — APPROVED** [review]
Really nice, thanks Athena!

**Review by abouquet27 — APPROVED** [review]
LGTM ! Well done !

**Review by abouquet27 — APPROVED** [review]
LGTM ! Well done !

**Pull request merged**